src/requests/AuthApi.ts

```diff
@@ -0,0 +1,223 @@
+import { VRCWrapper } from "../VRCWrapper";
+import { ApiPaths } from "../types/ApiPaths";
+import { BaseApi } from "./BaseApi";
+import totp from 'totp-generator';
+
+
+
+
+export class AuthApi extends BaseApi {
+    baseClass: VRCWrapper;
+    regexCode: RegExp;
+
+    constructor(baseClass: VRCWrapper) {
+        super(baseClass);
+        this.baseClass = baseClass;
+        this.regexCode = new RegExp('^[0-9]{6}$');
+    }
+
+/**
+ * **Get Current User Info**
+ * 
+ * This method lets you get the information about the current user being logged in with the current instance.
+ * 
+ * Cookies needs to be valid for this to work. If the user isn't authenticated, then this will throw an error.
+ * @returns {Promise<VRCAPI.Auth.responseTypes.loginResponse>} - Returns a Promise with the response from the API.
+ */
+    async getCurrentUser(): Promise<VRCAPI.Users.Models.CurrentUser> {
+
+        const paramRequest: VRCAPI.Generics.executeRequestType = {
+            currentRequest: ApiPaths.auth.getCurrentUserInfo,
+            pathFormated: ApiPaths.auth.getCurrentUserInfo.path,
+        };
+
+        this.checkValidData(paramRequest);
+        const queryResult = await this.executeRequest<VRCAPI.Users.Models.CurrentUser>(paramRequest);
+
+        return queryResult;
+    }
+
+    /**
+     * Checks if a user by a given `username`, `displayName` or `email` exist. 
+     * 
+     * This is used during registration to check if a username has already been taken, during change of displayName to check if a displayName is available, and during change of email to check if the email is already used. 
+     * 
+     * In the later two cases the `excludeUserId` is used to exclude oneself, otherwise the result would always be true.
+     * 
+     * It is **REQUIRED** to include **AT LEAST** `username`, `displayName` or `email` query parameter.
+     * 
+     * Although they can be combined - in addition with `excludeUserId` (generally to exclude yourself) - to further fine-tune the search.
+     */
+    async userExist({ email,
+        displayName,
+        userId,
+        excludeUserId }: VRCAPI.Auth.Requests.checkUserExistOptions): Promise<VRCAPI.Auth.responseTypes.checkUserExistResponse> {
+
+        const parameters: URLSearchParams = new URLSearchParams();
+
+        if (!email && !displayName && !userId) {
+            throw new Error("No search term was provided! Please provide at least one of the following: `email`, `displayName` or `userId`.");
+        }
+        console.log(excludeUserId);
+
+        if (email) {
+            parameters.append('email', email);
+        }
+
+        if (displayName) {
+            parameters.append('displayName', displayName);
+        }
+
+        if (userId) {
+            parameters.append('userId', userId);
+        }
+
+        if (excludeUserId) {
+            parameters.append('excludeUserId', excludeUserId);
+        }
```

Comment:
Optional: You can probably do these on a loop.

Reply:
Fixed